test/libwhisper/bloomFilter.cpp

```diff
@@ -244,4 +244,53 @@ BOOST_AUTO_TEST_CASE(bloomFilterRaw)
 	BOOST_REQUIRE(!f.contains(b00110111));
 }
 
-BOOST_AUTO_TEST_SUITE_END()
\ No newline at end of file
+static const unsigned DistributionTestSize = 8;
+
+void updateDistribution(FixedHash<DistributionTestSize> const& _h, unsigned* _distribution)
+{
+	unsigned bits = 0;
+	for (unsigned i = 0; i < DistributionTestSize; ++i)
+		if (_h[i])
+			for (unsigned j = 0; j < 8; ++j)
+				if (_h[i] & c_powerOfTwoBitMmask[j])
+				{
+					_distribution[i * 8 + j]++;
+					if (++bits >= TopicBloomFilterTest::BitsPerBloom)
+						return;
+				}
+}
+
+BOOST_AUTO_TEST_CASE(distributionRate)
+{
+	cnote << "Testing Bloom Filter Distribution Rate...";
+
+	unsigned const c_arrSize = DistributionTestSize * 8;
+	unsigned distribution[c_arrSize];
+	memset(distribution, 0, c_arrSize * sizeof(unsigned));
```

Comment:
Why not array of unsigned?  Default allocator will set values to 0 and (typically) ensures for loop won't go out of bounds.